docs/source/refs/reference_architecture/index.rst

```diff
@@ -0,0 +1,361 @@
+
+Most RL libraries expect their own variation of an environment interface. This means the
+data types needed by each library differs. Isaac Lab provides its own wrappers to convert
+the environment into the expected interface by the RL library a user wants to use. These are
+specified in the `Isaac Lab utils wrapper module <https://isaac-sim.github.io/IsaacLab/main/source/api/lab_tasks/omni.isaac.lab_tasks.utils.wrappers.html#module-omni.isaac.lab_tasks.utils.wrappers>`__.
+
+See the `full list <https://gymnasium.farama.org/api/wrappers/#gymnasium.Wrapper>`__ of other wrappers APIs. For more information on how these wrappers work,
+please refer to the `Wrapping environments <https://isaac-sim.github.io/IsaacLab/main/source/how-to/wrap_rl_env.html#how-to-env-wrappers>`__ documentation.
+
+**Adding your own wrappers**
+
+You can define your own wrappers by adding them to the Isaac Lab utils wrapper module. More information is available `on the GitHub page for wrapping environments <https://isaac-sim.github.io/IsaacLab/main/source/how-to/wrap_rl_env.html#adding-new-wrappers>`__.
+
+**Component 6 - Run Training**
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Finally, the last step is to run the training of the RL agent. Isaac Lab provides scripts which utilizes four popular RL libraries for training the models (GPU-based training):
+
+* `StableBaselines3 <https://stable-baselines3.readthedocs.io/en/master/>`__
+* `RSL-RL <https://github.com/leggedrobotics/rsl_rl>`__
+* `RL-Games <https://github.com/Denys88/rl_games>`__
+* `SKRL <https://skrl.readthedocs.io/en/latest/>`__
+
+
+.. note::
+
+  Isaac Lab does not provide the implementation of these RL libraries. They are already implemented by different authors. We provide the environments and framework wrappers for the RL libraries.
+
+
+
+If you want to integrate a different version of the provided algorithms or your learning library, you can follow
+`these instructions <https://isaac-sim.github.io/IsaacLab/main/source/how-to/add_own_library.html>`__.
+
+
+
+**Single GPU Training**
+
+.. image:: ../../_static/reference-architecture/single-gpu-training-light.svg
+    :class: only-light
+    :align: center
+    :alt: Single GPU Training Data Flow
+
+.. image:: ../../_static/reference-architecture/single-gpu-training-dark.svg
+    :class: only-dark
+    :align: center
+    :alt: Single GPU Training Data Flow
+
+Isaac Lab supports training massively parallel environments to speed up RL training and provides rich data for the model to train.
+For single GPU training, the following steps show how training works in Isaac Sim and Isaac Lab:
+
+1. Isaac Sim provides the asset states such as robot and sensor states, including the observations defined in the task observation config class.
+
+2. In Isaac Lab
+
+  * We add randomization to the states defined in the event configuration class to obtain the observation for the task.
```

Comment:
is it possible to format these bullet points nicer as well? right now, each bullet section has a box wrapped around it

Reply:
done